#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
from os.path import abspath, basename, dirname, splitext

PACKAGE = basename(dirname(dirname(abspath(__file__))))

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, default=True, description="Whether to apply this plugin or not")
gen.add("network_deviation_distance", double_t, 0, default=1.5,
        description="Euclidean distance of allowed deviation from network")
gen.add("plan_deviation_distance", double_t, 0, default=1.5,
        description="Euclidean distance of allowed deviation from plan")
gen.add("network_param_path", str_t, 0, default="~/network_planner/planner/graphs",
        description="Path of ROS parameter containing network")

exit(gen.generate(PACKAGE, PACKAGE + "_node", splitext(basename(__file__))[0]))
